src/main/java/com/appoptics/integrations/kafka/broker/KafkaAppopticsReporter.java

```diff
@@ -25,10 +28,33 @@ public synchronized void init(VerifiableProperties props) {
         }
 
         String token = props.getString("appoptics.token");
-        String source = props.getString("appoptics.agent.identifier");
+        List<Tag> tags = new ArrayList<Tag>();
+
+        String source = props.getString("appoptics.agent.identifier", "");
+        if (!source.equals("")) {
```

Comment:
[`isEmpty()`](https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#isEmpty())? Ditto in otherplaces.